pkg/sources/kafka/reader.go

```diff
@@ -324,18 +324,21 @@ func NewKafkaSource(
 		}
 	}
 	kafkasource.config = config
-	// Best effort to initialize the clients for pending messages calculation
-	adminClient, err := sarama.NewClusterAdmin(kafkasource.brokers, config)
+	client, err := sarama.NewClient(kafkasource.brokers, config)
 	if err != nil {
-		kafkasource.logger.Warnw("Problem initializing sarama admin client", zap.Error(err))
+		return nil, fmt.Errorf("failed to create sarama client, %w", err)
 	} else {
-		kafkasource.adminClient = adminClient
+		kafkasource.saramaClient = client
 	}
-	client, err := sarama.NewClient(kafkasource.brokers, config)
+	// Does it require any special privileges to create a cluster admin client?
+	adminClient, err := sarama.NewClusterAdminFromClient(client)
```

Comment:
Reference to https://github.com/kedacore/keda/blob/cf3cd1dc49bd77e86d57e8a7b12689f70dd91348/pkg/scalers/kafka_scaler.go#L393-L399